torch/_tensor_docs.py

```diff
@@ -3733,6 +3733,36 @@ def callable(a, b) -> number
 ``x.T`` is equivalent to ``x.permute(n-1, n-2, ..., 0)``.
 """)
 
+add_docstr_all('real',
+               r"""
+Returns a new tensor containing real values of the :attr:`self` tensor.
+The returned tensor shares the same data as :attr:`self`. For complex tensors,
+the returned tensor has the corresponding real dtype.
+
+Example::
+    >>> x=torch.randn(4, dtype=torch.cfloat)
+    >>> x
+    tensor([(0.3100+0.3553j), (-0.5445-0.7896j), (-1.6492-0.0633j), (-0.0638-0.8119j)])
+    >>> x.real
+    tensor([ 0.3100, -0.5445, -1.6492, -0.0638])
+
+""")
+
+add_docstr_all('imag',
+               r"""
+Returns a new tensor containing imaginary values of the :attr:`self` tensor.
+The returned tensor shares the same data as :attr:`self`. For complex tensors,
+the returned tensor has the corresponding real dtype.
```

Comment:
Should probably make clearer that this only works on complex tensors. You could add a warning that it's not implemented for non-complex tensors, or change the language to be something like:

"Returns a view of the complex tensor's imaginary values..."